### ISSUE TITLE
Questions about how to run

### ISSUE THREAD

**Issue by SunAnLan** [Zehaos/MobileNet#2]
I saw many models in the folder nets. Seems most of  them have a corresponding test.py . What are they testing?Seems its using some random array to run forward.Is it testing speed ? If so why is it gonna take 3~5 s for a batch that's quite slow.Hope for help

**Comment by Zehaos** [comment]
It is unit testing which is used for testing the functionality. I don't know weather it is a standard to test the speed.

**Comment by SunAnLan** [comment]
soga, its quite slow and I didn't find where you test the speed?Could you please show me that part code?Hope for you help @Zehaos

**Comment by Zehaos** [comment]
嗖得丝奈～ You can use python 'time' package or [timeline](https://github.com/tensorflow/tensorflow/issues/1824), it seems to be a more powerful tools, but I have no knowledge about it yet.

**Comment by kwotsin** [comment]
@SunAnLan It probably depends on your batch size and whether you're using CPU or GPU. The implementation is at least 3x faster in training than an inception v3 for me.

**Comment by SunAnLan** [comment]
纳入后头~洪洞阿里噶多 @Zehaos

**Comment by shicai** [comment]
因吹斯听。。。

**Comment by Zehaos** [comment]
@kwotsin , Could you share your equipment and the inference time? Thanks.

**Comment by SunAnLan** [comment]
Did you try to run forward ？ For me it’s quite slow？ @kwotsin

**Comment by Zehaos** [comment]
@SunAnLan 

You can measure inference time in this function [testForward](https://github.com/Zehaos/MobileNet/blob/master/nets/mobilenet_test.py#L68)
```
import time

start_time = time.time()
output = sess.run(logits)
duration = time.time() - start_time
```

**Comment by shicai** [comment]
@Zehaos what is the total training time to train this model?
how many hours for one epoch?
As for me, it is very very slow.

**Comment by SunAnLan** [comment]
@Zehaos  Yeah, that's exactly what I am doing now. and through I got the forward speed is about 0.9s on GTX 1080 8G

**Comment by Zehaos** [comment]
@shicai Using 2 GTX1080 in asynchronous mode, it spends 14d 14h 20min totally, about 2.8h per epoch.
Have you ever train other network on imagenet before? I have no concept about the training time.

**Comment by Zehaos** [comment]
@SunAnLan You'd better run it for several  iteration, and then compute the average. 
Looking forward to your result!

**Comment by Zehaos** [comment]
@SunAnLan I got 0.09s using a full-load gtx1060.... May be this is not a right way to benchmark.

**Comment by Zehaos** [comment]
I got 0.004 +/- 0.000 sec per image, using [time_benchmark.py](https://github.com/Zehaos/MobileNet/blob/master/time_benchmark.py).

**Comment by SunAnLan** [comment]
Got it I run 100 iteration and the average is about 0.09S per pic.This seems works!

**Comment by SunAnLan** [comment]
I also got 0.09s per pic now . average after 100 iteration

**Comment by Zehaos** [comment]
@SunAnLan Use [time_benchmark.py](https://github.com/Zehaos/MobileNet/blob/master/time_benchmark.py) instead.

**Comment by SunAnLan** [comment]
get ! @Zehaos

**Comment by austingg** [comment]
@Zehaos  @shicai  It is quite slow. 
 I haven't use tensorflow for imagenet training, However each epoch costs about 35 mins when training resnet18 using MxNet (two gtx 1080). As for mobilenet, it costs about 2h using mxnet, Since currently the separate conv2d is implemented by group.

**Comment by Zehaos** [comment]
@austingg @shicai I have do a benchmark on backward speed [README](https://github.com/Zehaos/MobileNet#time-benchmark), it seems that the backward speed of mobilenet is quite slow. Backward time is almost 4 times as the forward time.
[CNN-Benchmarks](https://github.com/jcjohnson/cnn-benchmarks)

**Comment by kwotsin** [comment]
@Zehaos I ran my model training on an NVIDIA GTX860M, and my image inference time is roughly 0.85s for an image of size 299x299, when I froze my graph.

**Comment by Zehaos** [comment]
@kwotsin Could you show me your code?

**Comment by kwotsin** [comment]
@Zehaos No problem. Here is the gist: https://gist.github.com/kwotsin/292eb12600be02b75bf69ff8010d07ce

I also included the showing of the image but the run time is computed before the image is shown. Btw do you have a rough timeline of when your pre-trained imagenet model for mobilenet will be out? Looking forward to it! :D

**Comment by Zehaos** [comment]
@kwotsin Why not start the timer exactly before fetching the prediction?
The pretrained weights is now available [README](https://github.com/Zehaos/MobileNet#accuracy-on-imagenet-2012-validation-set)

**Comment by kwotsin** [comment]
Hmm I'm not very sure how the inference time is measured, so I simply timed from the start till the end. I was thinking if I were to predict an image, I'll probably need to construct the graph beforehand also which would take some time, so I included the graph construction time as well.

**Comment by SunAnLan** [comment]
I tried your time_benchmark.py. I got 0.063 +/- 0.006 sec forward on my GTX 1080 8G. How did you make it that fast??? @Zehaos

**Comment by Zehaos** [comment]
@SunAnLan [time_benchmark.py#L36](https://github.com/Zehaos/MobileNet/blob/master/time_benchmark.py#L36) Use gpu instead...

**Comment by SunAnLan** [comment]
My fault!....2333 @Zehaos

**Comment by Satisfie** [comment]
Hi, when i use engine:CAFFE, it's real slow, I use 8 TitanXP, batchsize=1024, 10s per iter, real slow.
other model like inception-resnet-v2 just use 1s per iter
